src/test/java/io/strimzi/kafka/bridge/mqtt/core/MqttKafkaMapperTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.bridge.mqtt.core;
+
+import io.strimzi.kafka.bridge.mqtt.utils.MappingRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertThrows;
+
+/**
+ * Unit tests for {@link MqttKafkaMapper}
+ */
+public class MqttKafkaMapperTest {
+
+
+    /**
+     * Test for default topic.
+     * If the MQTT topic does not match any of the mapping rules, the default topic is used.
+     * E.g. if the Topic Mapping Rules is empty, the default topic is used by default.
+     */
+    @Test
+    public void testDefaultTopic() {
+        List<MappingRule> rules = new ArrayList<>();
+
+        MqttKafkaMapper mapper = new MqttKafkaMapper(rules);
+
+        assertThat("Should use the default topic when no mapping pattern matches.",
+                mapper.map("sensor/temperature"), is(MqttKafkaMapper.DEFAULT_KAFKA_TOPIC));
```

Comment:
well that's an example of just no rules ... but what when there are some rules and they don't match, for any reasons? see my next comment related to the wrong placeholder name, that could be another use case.

Reply:
The behavior would be just the same. No matches will be found.